libbpf-rs/src/ringbuf.rs

```diff
@@ -0,0 +1,146 @@
+use core::ffi::c_void;
+use std::boxed::Box;
+use std::ptr;
+use std::slice;
+use std::time::Duration;
+
+use crate::*;
+
+/// The canonical interface for managing a collection of [`RingBuffer`]s.
+///
+/// `ringbuf`s are a special kind of [`Map`], used to transfer data between
+/// [`Program`]s and userspace.  As of Linux 5.8, the `ringbuf` map is now
+/// preferred over the `perf buffer`.
+pub struct RingBufferManager<F>
+where
+    F: FnMut(&[u8]) -> i32 + 'static,
+{
+    ptr: *mut libbpf_sys::ring_buffer,
+    ringbufs: Vec<Box<F>>,
+}
+
+impl<F> RingBufferManager<F>
```

Comment:
we have `PerfBuffer`, so let's stick to `RingBuffer` here? It also matches libbpf naming.

Reply:
Sure, agreed